### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by pbernasconi**
@jabillings this includes an `authedAgent` to include before running each test.

### CODE REVIEW

File: test/routes/favorite.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+describe('/api/schedule', function() {
+
+  before(Promise.coroutine(function*(done) {
+    authedAgent = yield agent;
+  }));
+
+  it('GET / ', function(done) {
+    authedAgent
+      .get('/api/schedule')
```
Comment (danasilver) [inline test/routes/favorite.js:20]: Should these be `/api/favorites`?

File: test/routes/extracurricular.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+describe('/api/schedule', function() {
+
+  before(Promise.coroutine(function*(done) {
+    authedAgent = yield agent;
+  }));
+
+  it('GET / ', function(done) {
+    authedAgent
+      .get('/api/schedule')
```
Comment (danasilver) [inline test/routes/extracurricular.js:20]: Should these be `/api/extracurricular`?

Comment (pbernasconi) [inline test/routes/extracurricular.js:20]: yes - I just created the base, can complete tomorrow tho

**Comment by jabillings** [comment]
Great, thanks for doing this. I'll incorporate it into tests for the catalog and favorites API

**Pull request merged**